### ISSUE TITLE
Question about using ContrastiveLoss for modeling training

### ISSUE THREAD

**Issue by weikunhan** [KevinMusgrave/pytorch-metric-learning#295]
Hello Kevin, 

I recently fine-tune the model using ContrastiveLoss but looks like the model could not get coverage. And I believe I didn't set up ContrastiveLoss correctly. I posted my configuration for model training and training log. May I ask which part you think need to be adjusted? 

Model (the last layer apply L2 norm):
![image](https://user-images.githubusercontent.com/22847748/113766367-2c920e00-96d2-11eb-856e-5897417b0f90.png)

Pipeline (using RMSprop, DotProductSimilarity, and PairMarginMiner):
![image](https://user-images.githubusercontent.com/22847748/113766511-5c411600-96d2-11eb-9336-678e22874603.png)
**learning rate is 10^-5, weight decay 10^-4, momentum: 0.9**

Logs
The pre-trained model is 

|precision top 1 | precision top 5 | precision top 10 |
| --| --| --|
13.79%  | 35.63% | 41.38% |

![image](https://user-images.githubusercontent.com/22847748/113766736-a0341b00-96d2-11eb-9bc2-b2c7d29e4657.png)
![image](https://user-images.githubusercontent.com/22847748/113766755-a6c29280-96d2-11eb-94c2-ac87590ddff9.png)
**This number is indicated how many useful training samples have been mined by PairMarginMiner**
![image](https://user-images.githubusercontent.com/22847748/113766792-b2ae5480-96d2-11eb-86a8-2a4701d5908b.png)
![image](https://user-images.githubusercontent.com/22847748/113766829-bfcb4380-96d2-11eb-8e4c-8c5b96baabb6.png)
![image](https://user-images.githubusercontent.com/22847748/113766843-c48ff780-96d2-11eb-9221-77cd3582be48.png)

Thank you,

**Comment by KevinMusgrave** [comment]
It looks like you're using ```DotProductSimilarity``` as the ```distance``` for the contrastive loss, so you should change the ```pos_margin``` and ```neg_margin```, because [the defaults are suitable only for ```LpDistance```](https://kevinmusgrave.github.io/pytorch-metric-learning/losses/#contrastiveloss).

With a similarity metric, you should set the ```pos_margin``` to be bigger than the ```neg_margin```. For example:
```python
losses.ContrastiveLoss(pos_margin=1, neg_margin=0, distance=distance):
```

**Comment by weikunhan** [comment]
@KevinMusgrave 

Thank you so much, Kevin! 

I was wonder why I need to invert that two-parameter before.

Today, I took some time to review math and check your source code. Now, I know why it needs to invert it. 

Best, 

Mark
